src/Chainweb/Chainweb/MinerResources.hs

```diff
@@ -77,14 +77,15 @@ withMiningCoordination logger enabled cutDb inner
         fmap snd . concurrently (prune t) $ inner . Just $ MiningCoordination
             { _coordLogger = logger
             , _coordCutDb = cutDb
-            , _coordState = t }
+            , _coordState = t
+            , _coordLimit = 10000 }
```

Comment:
```bash
curl "https:/us1.tn1.chainweb.com/chainweb/0.0/development/chain/0/header?limit=100&minheight=3250" -H "accept:application/json;blockheader-encoding=object" -s | aeson-pretty | grep payloadHash | sed 's/.*"\(.*\)".*/\1/' | while read a ; do curl "https://us1.tn1.chainweb.com/chainweb/0.0/development/chain/0/payload/$a" -s | wc -c ; done | sed 's/^ *//'
```

This gives an avg payload size (JSON encoded, binary data base base64) of about 90kb. So 10,000 items would result tin about 500-1000MB.

In don't think we should store that much in a `Data.Map`.

I propose, that we reduce the size drastically, and add logging that sends telemetry about its size, so that we get an idea what actual requirements are.

Reply:
Generally, we should store one payload per open connection. We should investigate how we can do that.

For instance, in order to mine successfully, a client keeps an HTTP connection to the server open. Can we use this as a trigger? if the connection goes away, throw away all payloads that are associated with the connection. 

Another option would be to use RPC style communication (e.g. web sockets, HTTP2, or bare TCP sockets) and to associate the payload with the connection. E.g. each connection would have its own thread and the current payload is stored in the context of that thread.

The classical solution to store session data, is a Redis cache. But let's try to avoid going down that route.

 (Actually, should be possible to store only one set of transactions per chain and share that across all miner specific payloads. But that seems like a future optimization option.)

Reply:
The current state still is a big DOS vector.